x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/DataFrameAnalyticsManagerIT.java

```diff
@@ -0,0 +1,410 @@
+        Assert.assertThat(searchResponse.getHits().getTotalHits().value, equalTo((long) 5));
+        for(SearchHit hit : searchResponse.getHits().getHits()) {
+            Map<String, Object> src = hit.getSourceAsMap();
+            assertNotNull(src.get("outlier_score"));
+            assertThat(src.get("outlier_score"), equalTo(EXPECTED_OUTLIER_SCORE));
+        }
+
+        verify(task, never()).markAsFailed(any(Exception.class));
+        verify(task, times(1)).markAsCompleted();
+    }
+
+    public void testTaskContinuationFromReindexStateWithPreviousResultsIndex() throws Exception {
+        internalCluster().ensureAtLeastNumDataNodes(1);
+        ensureStableCluster(1);
+        String sourceIndex = "test-outlier-detection-from-reindex-state-with-results";
+        createIndexForAnalysis(sourceIndex);
+        String id = "test_outlier_detection_from_reindex_state_with_results";
+        DataFrameAnalyticsConfig config = buildOutlierDetectionAnalytics(id, sourceIndex);
+        putDataFrameAnalyticsConfig(config);
+
+        // Create the "results" index, as if we ran reindex already in the process, but did not transition from the state properly
+        createAnalysesResultsIndex(config.getDest(), false);
+        List<AnalyticsResult> results = buildExpectedResults(sourceIndex);
+
+        DataFrameAnalyticsManager manager = createManager(results);
+
+        DataFrameAnalyticsTask task = buildMockedTask(config.getId());
+        manager.execute(task, DataFrameAnalyticsState.REINDEXING);
+
+        // wait for markAsCompleted() or markAsFailed() to be called;
+        assertBusy(() -> assertTrue(isCompleted()), 120, TimeUnit.SECONDS);
+
+        // Check we've got all docs
+        SearchResponse searchResponse = client().prepareSearch(config.getDest()).setTrackTotalHits(true).get();
+        Assert.assertThat(searchResponse.getHits().getTotalHits().value, equalTo((long) 5));
+        for(SearchHit hit : searchResponse.getHits().getHits()) {
+            Map<String, Object> src = hit.getSourceAsMap();
+            assertNotNull(src.get("outlier_score"));
+            assertThat(src.get("outlier_score"), equalTo(EXPECTED_OUTLIER_SCORE));
+        }
+
+        verify(task, never()).markAsFailed(any(Exception.class));
+        verify(task, times(1)).markAsCompleted();
+    }
+
+    public void testTaskContinuationFromAnalyzeState() throws Exception {
+        internalCluster().ensureAtLeastNumDataNodes(1);
+        ensureStableCluster(1);
+        String sourceIndex = "test-outlier-detection-from-analyze-state";
+        createIndexForAnalysis(sourceIndex);
+        String id = "test_outlier_detection_from_analyze_state";
+        DataFrameAnalyticsConfig config = buildOutlierDetectionAnalytics(id, sourceIndex);
+        putDataFrameAnalyticsConfig(config);
+        // Create the "results" index, as if we ran reindex already in the process, but did not transition from the state properly
```

Comment:
In this case we did transition properly to analyzing

Reply:
comment copy-pasta :).